apps/mobile-wallet/src/features/buy/BuyModal.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { NavigationProp, useNavigation } from '@react-navigation/native'
+import { useEffect, useRef, useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import { BackHandler, Platform } from 'react-native'
+import { Portal } from 'react-native-portalize'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import WebView, { WebViewNavigation } from 'react-native-webview'
+import styled, { useTheme } from 'styled-components/native'
+
+import AppText from '~/components/AppText'
+import Button from '~/components/buttons/Button'
+import BottomModal, { BottomModalProps } from '~/components/layout/BottomModal'
+import { ModalContent } from '~/components/layout/ModalContent'
+import ScreenTitle from '~/components/layout/ScreenTitle'
+import LinkToWeb from '~/components/text/LinkToWeb'
+import { useAppSelector } from '~/hooks/redux'
+import { InWalletTabsParamList } from '~/navigation/InWalletNavigation'
+import { selectDefaultAddress } from '~/store/addressesSlice'
+import { DEFAULT_MARGIN } from '~/style/globalStyle'
+
+interface BuyModalProps extends Omit<BottomModalProps, 'Content'> {}
+
+const BuyModal = (props: BuyModalProps) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<NavigationProp<InWalletTabsParamList>>()
+  const theme = useTheme()
+  const webViewRef = useRef<WebView>(null)
+  const insets = useSafeAreaInsets()
+  const defaultAddress = useAppSelector(selectDefaultAddress)
+  const [isDisclaimerAccepted, setIsDisclaimerAccepted] = useState(false)
+
+  const onAndroidBackPress = () => {
+    if (webViewRef.current) {
+      webViewRef.current.goBack()
+      return true // prevent default behavior (exit app)
+    }
+    return false
+  }
```

Comment:
nit: I would move the declaration of this function under line 59 (inside the condition of the `useEffect` so that a new function is not declared on every single rerender, but only once, when the modal is opened, and if the platform is Android.

Reply:
Done